src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,106 @@
+package duke.ui;
+
+import javafx.scene.image.Image;
+
+/**
+ * Ui is a utility class for reading and writing input to the Duke program.
+ * <p/>
+ * Ui is a wrapper around BufferedReader and PrintWriter that reads from System.in InputStream
+ * and writes to System.out OutputStream.
+ * </p>
+ * An Ui instance also stores the message to be shown before showResponse flushes the message to the outputsteam.
+ */
+public class Ui {
+    public static final String BYE_MESSAGE = "Roarrr....Let's burn more tasks next time!";
+    public static final String ADD_MESSAGE = "Charizard is ready to burn task:";
+    public static final String REMOVE_MESSAGE = "Charizard got tired of waiting and deleted this task";
+    public static final String MARK_MESSAGE = "Charizard breathe out fire and burned the task.";
+    public static final String UNMARK_MESSAGE = "Oh no! The task was not burnt completely!";
+    public static final Image GENERAL_IMAGE = new Image(Ui.class.getResourceAsStream("/images/General.png"));
+
+    private static final int BORDER_LENGTH = 38;
+    private static final String GREET_MESSAGE = "Roarrr.... I'm Burning Charizard, tasked to burnnn down your tasks.\n"
+            + "Which task shall we burn today?";
+
+    private static final Image ADD_OR_DELETE_IMAGE = new Image(Ui.class.getResourceAsStream("/images/Add.png"));
+    private static final Image MARK_IMAGE = new Image(Ui.class.getResourceAsStream("/images/Mark.png"));
+    private static final Image UNMARK_IMAGE = new Image(Ui.class.getResourceAsStream("/images/Unmark.PNG"));
+    private static final Image ERROR_IMAGE = new Image(Ui.class.getResourceAsStream("/images/Error.PNG"));
+    private Image responseImage = GENERAL_IMAGE;
+    private StringBuilder message;
+
+    /**
+     * Constructs a new Ui instance.
+     */
+    public Ui() {
+        message = new StringBuilder();
+    }
+
+    /**
+     * Appends the message to the stored message in Ui.
+     *
+     * @param message The message to be appended.
+     */
+    public void appendMessage(String message) {
+        this.message.append(message);
+    }
+
+    /**
+     * Appends the line border to the stored message in Ui.
+     */
+    public void appendBorder() {
+        appendMessage("\n");
+        for (int i = 0; i < BORDER_LENGTH; i++) {
+            appendMessage("-");
+        }
+        appendMessage("\n");
+    }
```

Comment:
Perhaps you might want to consider using `System.lineSeparator()` because on Windows a new line is `\r\n` while on Unix it is `\n`. A helpful way to implement it is to declare it as a constant such as `public static final String LS = System.lineSeparator();` along with your messages above.